daemon/cmd/state.go

```diff
@@ -92,8 +93,16 @@ func (d *Daemon) validateEndpoint(ep *endpoint.Endpoint) (valid bool, err error)
 }
 
 func (d *Daemon) getPodForEndpoint(ep *endpoint.Endpoint) error {
-	d.k8sWatcher.WaitForCacheSync(resources.K8sAPIGroupPodV1Core)
-	pod, err := d.k8sWatcher.GetCachedPod(ep.K8sNamespace, ep.K8sPodName)
+	var (
+		pod *slim_corev1.Pod
+		err error
+	)
+	if option.Config.EnableHighScaleIPcache {
+		pod, _, _, _, _, err = d.fetchK8sMetadataForEndpoint(ep.K8sNamespace, ep.K8sPodName)
+	} else {
+		d.k8sWatcher.WaitForCacheSync(resources.K8sAPIGroupPodV1Core)
```

Comment:
Maybe @cilium/sig-k8s folks could speak to this more, but I was under the impression that the Pod watcher only listens for local Pods anyway, which would make these changes moot? :thinking:

Reply:
cc @christarazi

Reply:
See https://github.com/cilium/cilium/pull/25148#discussion_r1189229258.